### ISSUE TITLE
Unable to swap subs to Orca 

### ISSUE THREAD

**Issue by phroureo** [Regalis11/Barotrauma#7814]
We started in the Blue Guppy (a modded sub), and then bought the Imp (another modded sub). After that, we bought the Orca 2 and were able to use it without problem. 

When we tried to swap BACK to the Orca 2, it would not let us. We were able to switch between the Guppy and the Imp without issue. 

My friend tried to reproduce the issue in a single player campaign and got some errors that may be related.

https://cdn.discordapp.com/attachments/162699457828356097/921211503963697163/unknown.png

https://cdn.discordapp.com/attachments/162699457828356097/921211576114110505/unknown.png

**Comment by Regalis11** [comment]
>My friend tried to reproduce the issue in a single player campaign and got some errors that may be related.

Would it be possible to send us a screenshot of those errors?

**Comment by phroureo** [comment]
There are two discord links in the original message with the errors that he saw. Let me know if you need me to upload them on imgur or something instead.

**Comment by Regalis11** [comment]
Ah, forgot to get back to this, sorry for the super later reply! :/

As I mentioned the screenshots are from a multiplayer campaign, not a single player one, so I'm afraid they don't shed much light into what happened.

Some more details on how exactly the swapping failed could also give some clues for diagnosing this. When you tried to swap, what happened exactly? Was there some error message, did the buttons not do anything, did the sub not get switch despite you selecting to switch it?

**Comment by phroureo** [comment]
Ah I see. Some of the details may be slightly fuzzy, but what I think
happened if I remember right:

We started a campaign, bought a second sub and switched to it, then a while
later bought a third sub and switched to it. When we tried to switch back
to one of the other subs (possibly the second, which may have been a
workshop sub iirc?) it would not let us. We could pick it from the switch
menu, but it just wouldn't pop up the voting option.

I *believe* that I read a patch notes saying that this or something similar
was fixed but I think out campaign was already broken so we gave up before
having a chance to test it. :/
